Assets/Mirror/Editor/Weaver/Processors/MessageClassProcessor.cs

```diff
@@ -1,11 +1,26 @@
 // this class generates OnSerialize/OnDeserialize when inheriting from MessageBase
+
+using System.Linq;
 using Mono.CecilX;
 using Mono.CecilX.Cil;
 
 namespace Mirror.Weaver
 {
     static class MessageClassProcessor
     {
+
+        static readonly OpCode[] emptyBodyTemplate = { OpCodes.Nop, OpCodes.Ret };
+
+        static bool IsEmptyDefault(this MethodBody body)
+        {
+            if (body.Instructions.Count != emptyBodyTemplate.Length) return false;
```

Comment:
how about:
```cs
return body.Instructions.All(instruction => instruction.OpCode == OpCodes.Nop || instruction.OpCode == OpCodes.Ret);
```

Reply:
possible, but not needed as empty is nop+ret in exact succession

Reply:
yeah,  but much simpler and readable

Reply:
I think that's debatable, All() says, everything is either nop or ret which is still somewhat vague about the intention. But if you want that to be a oneliner, SequenceEquals() should be descriptive enough.

Reply:
SequenceEquals is ok

Reply:
TBH this PR is a bit of a hack,  but it does make life easier.

One day unity will have C# 8,  which has default interface methods.   This will allow us to provide a default implementation of the methods right in the interface.   Our weaver will detect that and generate the methods.  Then we will be out of the business of trying to detect empty methods,  but for now,  I would not want to rely on that nop being there.

Reply:
probably, yes. But then again all this weaving is a hack in itself already for features the language does not natively support. And as unity is using roslyn i would expect this pattern to be somewhat stable.
But as you please, All() it is then.